File: backend/app/schemas/campaign.py
from pydantic import BaseModel

from app.schemas.common import PaginatedResponse


class CampaignSummary(BaseModel):
    id: int
    name: str
    users_count: int
    average_income: float


class CampaignUserOut(BaseModel):
    id: int
    original_id: int
    name: str
    age: int
    city: str
    income: float


class CampaignDetailResponse(BaseModel):
    campaign: CampaignSummary
    users: PaginatedResponse[CampaignUserOut]

File: frontend/src/types/campaign.ts
import { UserData } from "./user";
import { PaginatedResponse } from "./common";

export interface CampaignSummary {
  id: number;
  name: string;
  users_count: number;
  average_income: number;
}

export interface CampaignDetailResponse {
  campaign: CampaignSummary;
  users: PaginatedResponse<UserData>;
}
